airbyte-webapp/STYLEGUIDE.md

```diff
@@ -70,3 +70,111 @@ Our SCSS color variables compile to `rgb(X, Y, Z)`, which is an invalid value in
+
+* components - All React components except for page components. See section below for details
+* config - Config system
+* core - General modules used by the app
+* hooks - Shared hooks
+* locales - i18n files.
+* packages - Folder deprecated. Sub-folders will be moved out in the future
+* pages - React components that represents the app routes and pages. See section below
+* scss - Global SCSS files
+* services -
+* test-utils - Unit testing utilities
+* types - Shared TypeScript types and external module type definitions
+* utils - Shared utilities
+* views - Folder deprecated. Includes React components that will be moved to `components/` in the future
+
+## React Components
+
+> The Airbyte team is currently restructuring how components are organized in the codebase. Components currently in `src/views/` are being be migrated to `src/components/`. `src/components` is currently being reorganized. The new structure is described below and it's expected that new code follows the new structure.
+
+Most React components are in `src/components` but components that represent pages on the site are in `src/pages`. 
+
+### Components Structure
+
+The `src/components` folder is divided into sub-folders for each domain in the app such as `connection`, `source`, and `destination`. Core UI components (such as Buttons, tables, etc.) are all located in `src/components/ui`. Components that are shared across different domains but may not be part of the UI library are in `src/components/common`. Sub-folders must be written in `camelCase`.
+
+Within each sub-folder, there are folder for each major component. These folders are written in `PascalCase`, the same way a React component would be named. Within these folders there are a few files:
+
+* `index.ts` - Used to the main component and supporting functions or types to the app.
+* `{MainComponentName}.tsx` - The main component, also exported through `index.ts`
+* `{MainComponentName}.test.tsx` - The test file
+* `{MainComponentName}.module.scss` - The main component's style file
+* Files for additional components that support the main Component directly, using the same naming conventions as above.
+* `types.ts` (optional) - Types shared between different sub-components
+* `utils.ts` (optional) - Any utilities that support the components
+
+When using supporting components, the folder could become rather full of them. In those cases Components should be split into their own folders parallel to the main component, especially when the supporting component also needs to be broken down into multiple sub components.
+
+Here's a hypothetical example: The app has a streams panel with a lot of sub-components including a streams table. Instead of placing all the components under `src/components/connection/StreamsPanel`, the table should be broken out into it's own sub-folder as a child of the `connection` folder.
```

Comment:
```suggestion
Here's a hypothetical example: The app has a streams panel with a lot of sub-components including a streams table. Instead of placing all the components under `src/components/connection/StreamsPanel`, the table should be broken out into its own sub-folder as a child of the `connection` folder.
```